### PR TITLE
sys_status : add MAV_TYPE as a parameter

### PR DESCRIPTION
**Pull request by Kiwa21**
add MAV_TYPE as a parameter that can be changed with the yaml config file.

### CODE REVIEW

File: mavros/launch/apm_config.yaml [hunk 1]
```diff
@@ -8,6 +8,7 @@ startup_px4_usb_quirk: false
 # sys_status & sys_time connection options
 conn:
   heartbeat_rate: 1.0    # send hertbeat rate in Hertz
+  conn_heartbeat_mav_type: 18 # ONBOARD_CONTROLLER
```
Comment (vooon) [inline mavros/launch/apm_config.yaml]: Better to extend uas lib to be able to find MAV_TYPE by string.

File: mavros/src/plugins/sys_status.cpp [hunk 2]
```diff
@@ -512,6 +519,9 @@ class SystemStatusPlugin : public plugin::PluginBase
 	ros::ServiceServer rate_srv;
 	ros::ServiceServer mode_srv;
 
+	int conn_heartbeat_mav_type;
```
Comment (vooon) [inline mavros/src/plugins/sys_status.cpp]: Should be `MAV_TYPE`, or at least underlying type for that enum.

**Comment by Kiwa21** [comment]
@vooon Thanks for suggestion. I updated my PR to use MAV_TYPE and find by string.

File: mavros/src/lib/enum_to_string.cpp [hunk 3]
```diff
@@ -176,6 +176,20 @@ std::string to_string(MAV_TYPE e)
...
+			std::underlying_type<MAV_TYPE>::type rv = idx;
+			return static_cast<MAV_TYPE>(rv);
+		}
+	}
+
+	ROS_ERROR_STREAM_NAMED("uas", "TYPE: Unknown MAV_TYPE: " << mav_type);
+	return MAV_TYPE::GENERIC;
+}
+
 // [[[end]]] (checksum: 31488f5970b0f82b3efef71e32590bb6)
```
Comment (vooon) [inline mavros/src/lib/enum_to_string.cpp]: Do not change generated code.

Also i think it is better to add enum entry name to value function.
For that another array is needed, or changes in current (use std::tuple).

File: mavros/src/plugins/sys_status.cpp [hunk 4]
```diff
@@ -430,16 +431,23 @@ class SystemStatusPlugin : public plugin::PluginBase
...
 		nh.param("sys/disable_diag", disable_diag, false);
 
-		// rate parameter
+		// heartbeat rate parameter
 		if (nh.getParam("conn/heartbeat_rate", conn_heartbeat_d) && conn_heartbeat_d != 0.0) {
 			conn_heartbeat = ros::Duration(ros::Rate(conn_heartbeat_d));
 		}
 
+		// heartbeat mav type parameter
+		if (nh.getParam("conn/heartbeat_mav_type", conn_heartbeat_mav_type_str) && !conn_heartbeat_mav_type_str.empty()) {
```
Comment (vooon) [inline mavros/src/plugins/sys_status.cpp]: `.empty()` check is not necessary. Heartbeat rate check is needed because we can set 0 to disable heartbeats.

**Pull request closed without merging**